File: csharp/core/exception/UndoRedoTransactionIsNotStartedException.cs
using System;

namespace urakawa.exception
{
    /// <summary>
    /// Thrown when an operation is not valid because no undo/redo transaction has been started
    /// </summary>
    public class UndoRedoTransactionIsNotStartedException : CheckedException
    {
        /// <summary>
        /// Constructor setting the message of the exception
        /// </summary>
        /// <param localName="msg">The message</param>
        public UndoRedoTransactionIsNotStartedException(string msg)
            : base(msg)
        {
        }

        /// <summary>
        /// Constructor setting the message and inner <see cref="Exception"/> of the exception
        /// </summary>
        /// <param localName="msg">The message</param>
        /// <param localName="inner">The inner exception</param>
        public UndoRedoTransactionIsNotStartedException(string msg, Exception inner)
            : base(msg, inner)
        {
        }
    }
}

File: java/core/src/org/daisy/urakawa/undo/UndoRedoTransactionIsNotStartedException.java
package org.daisy.urakawa.undo;

import org.daisy.urakawa.exception.CheckedException;

/**
 * <p>
 * This exception is raised when trying to run an operation that is not
 * permitted while an undo-redo transition is not currently active (not
 * started).
 * </p>
 */
public class UndoRedoTransactionIsNotStartedException extends CheckedException {
	/**
	 * 
	 */
	private static final long serialVersionUID = -4845462825860674740L;
}
